### PR TITLE
End to end testing image build

### PR DESCRIPTION
**Pull request by matthewelwell**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

Adds workflow to exercise the all in one image as an API (using the E2E testing image)

## How did you test this code?

New Github workflow.

### CODE REVIEW

**Comment by dabeeeenster** [comment]
When I run it on the EC2 instance, the tests run but I get a random failure, usually around the same spot in the tests (but not exactly the same each time). 

```
frontend_1       |   Create Features
frontend_1       |
frontend_1       | Waiting element visible #features-link
frontend_1       | Waiting element visible #show-create-feature-btn
frontend_1       | Waiting element visible #features-link
frontend_1       | Waiting element visible #show-create-feature-btn
frontend_1       | Waiting element visible #show-create-feature-btn
frontend_1       | Set text [data-test="featureID"] : flag
frontend_1       | Set text [data-test="featureDesc"] : description
frontend_1       | Waiting element visible [data-test="create-feature-btn"]
frontend_1       |  ✖ Submit a Form
frontend_1       |
frontend_1       |    1) AssertionError: expected false to be truthy
frontend_1       |
frontend_1       |       Browser: Chrome 96.0.4664.110 / Linux 0.0
frontend_1       |
frontend_1       |          10 |        .typeText(selector, `${text}`);
frontend_1       |          11 |};
frontend_1       |          12 |
frontend_1       |          13 |export const waitForElementVisible = async (selector) => {
frontend_1       |          14 |    console.log(`Waiting element visible ${selector}`);
frontend_1       |        > 15 |    return t.expect(Selector(selector).visible).ok();
frontend_1       |          16 |};
frontend_1       |          17 |
frontend_1       |          18 |export const waitForElementNotExist = async (selector) => {
frontend_1       |          19 |    console.log(`Waiting element not visible ${selector}`);
frontend_1       |          20 |    return t.expect(Selector(selector).exists).notOk('', {
frontend_1       |       timeout: 10000 });
frontend_1       |
frontend_1       |          at <anonymous> (/srv/flagsmith/e2e/[helpers.cafe](https://helpers.cafe/).js:15:49)
frontend_1       |          at asyncGeneratorStep (/srv/flagsmith/e2e/[helpers.cafe](https://helpers.cafe/).js:1:1069)
frontend_1       |          at _next (/srv/flagsmith/e2e/[helpers.cafe](https://helpers.cafe/).js:1:1407)
frontend_1       |          at <anonymous> (/srv/flagsmith/e2e/[helpers.cafe](https://helpers.cafe/).js:1:1572)
frontend_1       |          at <anonymous> (/srv/flagsmith/e2e/[helpers.cafe](https://helpers.cafe/).js:1:1313)
frontend_1       |          at waitForElementVisible
frontend_1       |       (/srv/flagsmith/e2e/[helpers.cafe](https://helpers.cafe/).js:13:35)
frontend_1       |          at <anonymous>
frontend_1       |       (/srv/flagsmith/e2e/cafe/[environment.cafe](https://environment.cafe/).js:19:11)
frontend_1       |          at asyncGeneratorStep
frontend_1       |       (/srv/flagsmith/e2e/cafe/[environment.cafe](https://environment.cafe/).js:1:156)
frontend_1       |          at _next (/srv/flagsmith/e2e/cafe/[environment.cafe](https://environment.cafe/).js:1:494)
frontend_1       |
```

**Comment by dabeeeenster** [comment]
This is working if I modify the docker-compose file to build the api:

```
    flagsmith-api:
        build:
            context: ../api
            dockerfile: Dockerfile

```

**Comment by dabeeeenster** [comment]
Waiting for @kyle-ssg to add a trailing `/` to this endpoint:

```
❯ curl 'http://localhost:8000/api/v1/environments/fD2SoyDihbQP8WyVGvjGMG' \
        -X 'PUT' \
```

File: frontend/Dockerfile.e2e [hunk 1]
```diff
@@ -3,7 +3,7 @@ FROM node:16
 RUN mkdir /srv/flagsmith && chown node:node /srv/flagsmith
 
 RUN apt-get update
-RUN wget https://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_96.0.4664.110-1_amd64.deb
+RUN wget -q https://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_96.0.4664.110-1_amd64.deb
```
Comment (dabeeeenster) [inline frontend/Dockerfile.e2e:6]: @kyle-ssg I think we have multiple places where the chrome version is defined? Is that right?
@matthewelwell  should move that out to an GH env var?

**Review by dabeeeenster — APPROVED** [review]

**Comment by dabeeeenster** [comment]
Approved but maybe we should have a canonical Chrome version somewhere?

**Comment by matthewelwell** [comment]
> Approved but maybe we should have a canonical Chrome version somewhere?

Yep, that makes sense but we should probably do that in a separate PR. I've created an issue for this [here](https://github.com/Flagsmith/flagsmith/issues/1748) and I'll merge this one for now.

**Pull request merged**